extra/modules/pb-request-correction/src/main/java/org/prebid/server/hooks/modules/pb/request/correction/core/correction/CorrectionProducer.java

```diff
@@ -0,0 +1,11 @@
+package org.prebid.server.hooks.modules.pb.request.correction.core.correction;
+
+import com.iab.openrtb.request.BidRequest;
+import org.prebid.server.hooks.modules.pb.request.correction.core.config.model.Config;
+
+public interface CorrectionProducer {
+
+    boolean shouldProduce(Config config, BidRequest bidRequest);
+
+    Correction produce();
```

Comment:
Please, change method definition a bit:
```
 Correction produce(Config config);
```

Also, if you have a good ideas about how to make it so produce will return `NoOp` and we can filter it out, it would be great (so we can get rid of `shouldProduce`).